### DISCUSSION TITLE
Smaller bundle for fully static deployments

### DISCUSSION THREAD

**Discussion by pablopunk** [vercel/next.js#15316] [Ideas]
Hi! So I've been using NextJS a lot and I even use it now whenever I just wanna build a static html/css small website. It's just really useful/easy/fast to build them with Next.

But let's say you build a completely static website. For example, this simple `pages/index.js`:

```js
export default function Index() {
  const now = new Date();
  return (
    <main>
      {now.toLocaleTimeString().slice(0, 5))}
    </main>
  );
}
```

It's just showing the time (`hh:mm`) once you visit the page. It doesn't really need react, or even dynamic updates. I deployed this example (with some styling) to [timenow.pablo.im](https://timenow.pablo.im) for the sake of testing.

If I do `curl -s https://timenow.pablo.im` I can see that the html is already rendered, cause I get `<main>12:06</main>` inside the body.

The "problem" is that this simple website is also including all of this code inside the body tag:

![Captura de pantalla 2020-07-20 a las 11 58 38](https://user-images.githubusercontent.com/4324982/87925319-6659b080-ca80-11ea-9ffe-461d21aaeebe.png)

Not only that, but inside the head tag you get even more code:

![Captura de pantalla 2020-07-20 a las 11 59 38](https://user-images.githubusercontent.com/4324982/87925388-7ec9cb00-ca80-11ea-8f34-0767bd7d9deb.png)

And it's performing all of these requests:

<img width="917" alt="Captura de pantalla 2020-07-20 a las 12 00 31" src="https://user-images.githubusercontent.com/4324982/87925482-a456d480-ca80-11ea-9c70-c4915099ed22.png">

## Question

I guess my question is: Are all of these requests and lines of code necessary? Is there a way to get rid of them? I mean maybe NextJS could be "smarter" about what a page needs or maybe we can specify it in the code.

Of course some of you might say "just do it with html/css" but it's not really my point here. Also, I know these simple websites are still really really fast. I was just wondering all of this cause I'm curious if we could get any benefits!

**Comment by jamesmosier** [comment]
Hi @pablopunk. There is currently no way to avoid those script imports today in Next.js and they are necessary for your site. But a "low code" feature of Next is something that is being considered. There is also a community plugin available that has some of these features today: https://github.com/amannn/next-client-script

**Comment by pablopunk** [comment]
Thanks for your answer @jamesmosier. It's good to know this is being considered and also thanks for sharing that project, it's really interesting.
